modules/ROOT/pages/revisionhistory.adoc

```diff
@@ -0,0 +1,165 @@
+
+include::partial$misc/admon-requires-7.0v.adoc[]
+
+include::partial$misc/admon-iframe-only.adoc[]
+
+The {pluginname} plugin provides users an ability to see how a document has changed over time and a simply to way to restore a saved version.
+
+== Interactive example
+
+liveDemo::{plugincode}[]
+
+
+=== How it works
+// TODO: Attach image of revision history
+
+The {pluginname} view is opened when clicking on the `revisionhistory` toolbar or menu button which locates under `View` menu.
+
+The view consists of these main UI components:
+
+* In the toolbar header, there are `Restore this version` and `Close` buttons. Clicking `Restore this version` will set the editor's content to the selected version and close the view. Pressing `Close` button will exit the view.
+
+* The readonly diff content UI shows the changes between the selected version and the closest one, which is below the selected version. The changes are highlighted for easy recognition of the types of change being applied:
+
+  ** Red: Removed content
+  ** Green: New content
+  ** Yellow: Content being modified. Modifications to HTML content implies attribute or format changes (bold, italic, etc)
+
+* The revisions sidebar renders the available versions for a document. When a new version is selected, the diff content UI is updated accordingly.
+
+[NOTE]
+The {pluginname} plugin treats commented HTML as actual content but the content is ignored in the diffing process. This means that a version with all its content commented appears empty in the diff content UI.
+
+=== Basic setup
+
+To setup the {pluginname} plugin user-interface in the editor:
+
+* add `{plugincode}` to the `plugins` option in the editor configuration;
+
+* add `{plugincode}` to the `toolbar` option in the editor configuration;
+
+* add `{plugincode}_fetch` option to the editor configuration;
+
+For example:
+
+[source,js]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'revisionhistory',
+  toolbar: 'revisionhistory',
+  // Required
+  revisionhistory_fetch: () => new Promise((resolve) => {
+    const revisions = [];  // Replace this with an API request to get saved versions
+    resolve(revisions);
+  })
+});
+----
+
+== Understanding initial and current versions
+
+A new document, that has no saved versions, always has at max two versions: the initial version and current version. 
+
+* Initial version is created on {productname} `Loaded` event with the content initialized with the editor.
+* Current version is added on opening the {pluginname} with the editor's current content. If included, it is always be the latest version and placed at the top.
+
+If both versions are the same, only the current version is kept.
+
+When a document has saved versions, its initial version is discarded as the {pluginname} assumes it already a part of versions. The current version is only discarded if it matches to the closest version.
```

Comment:
```suggestion
When dealing with a new document, that has no saved versions, {pluginname} plugin will always typically maintain two versions: the `initial` and `current` version.

* Initial Version: this version is generated during the {productname} `Loaded` event, with the editor's content initialized.
* Current Version: is added on opening the {pluginname} with the editor's current content. If included, it will become the `latest version` and placed at the top of the revisions list.

[TIP]
If both versions are the same, only the `current version` is kept.

For documents with saved versions, the `initial version` is disregarded by the {pluginname} assuming it's already part of the version history. The current version is only discarded if it matches the closest saved version.
```